src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs

```diff
@@ -160,7 +160,7 @@ public string Description
         [Parameter]
         public ProcessorArchitecture ProcessorArchitecture
         {
-            get { return _processorArchitecture.HasValue ? _processorArchitecture.Value : ProcessorArchitecture.None; }
+            get { return _processorArchitecture ?? ProcessorArchitecture.None; }
```

Comment:
I believe a preferred pattern is:
```suggestion
            get { return _processorArchitecture.GetValueOrDefault(ProcessorArchitecture.None); }
```
Below too.

Reply:
I prefer the first form, it is more readable.

Also, the JIT code quality seems better (in release):

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUCWAdjOQLIAUHMAPzlWASkbMGRZjMoB2EeUHCALDgDcEpgF8t5PWRGdyAOX6dhYvVNnMsC1gDoA4nBgA1CABsArnADyCAAicABmED5eMLxqoprSzLpE2kA==

Reply:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUCWAdjOQLIAUHMAPzlWASkbMGRZjMoB2EeUHCALDgDcEpgF8t5PWRGdyAOX6dhYvVNnMsC1gDoA4nBgA1CABsArnADyCAAicABmED5eMLxqoprSzLoJTAZsxv7mQiLiyfR6MvYijgAS0J6+cEpF5X7kIORq8TJJMqlGXAAKmZY5Mja28kWlUDWVli5uo4Eh4ZHRonUNGnpJ2kA==

Reply:
I am not sure what sharplab.io shows. You could look more real code:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUCWAdjOQLIAUHMAPzlWASkbMGRZjMoB2EeUHCALDgDcEpgF8t5PWRGdyAOX6dhYvVNnMsC1gDoA4nBgA1CABsArnADyCAAicABmED5eMLwqpKKa0sy6iUwGFFgq5AAq5kIi4in0ejLUAPbsUKVecI4A6gis8Lx8YvHFzGUVVTX1jXC8Zq0JMsnaQA=== - do you see a difference? :-)

Also it is not readable to hide a fact with "??" operator that the type is a _value_ type. Also  GetValueOrDefault() pattern is recommended.

Reply:
Interesting what happens when the type is changed to long:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUA2A9gHYDm5AsgAoOPAPzkAlgEpGzBkWYLKAdgnkRYgCw4AMgG4ZTAL4HyJsuWG8AckK7cxUk3MXMsK8QDoA4nBgA1CFYAVzgAeQQAETgAMwgg1hgBLW1JfXlmY3SmMzY7clDbUQlpLPoTBTcJDwAJaADguDUq+pDyEHJktIVMhRyLPIAFQvtip3LXdxq6wNaHb18WsMiYuISBSTaOnS6M4kMgA=

Reply:
Anyway, if GetValueOrDefault() pattern is recommended then we should submit an issue at dotnet/roslyn for the IDE0030 analyzer.

Reply:
Yes, it would be a subject for discussion there.

Reply:
@iSazonov I can't find documentation recommending `GetValueOrDefault (defaultValue)` over the null coalescing operator.

Reply:
I think it is a subject of the discussion https://github.com/dotnet/runtime/issues/33792

Reply:
I don't see any discussion of the null coalescing operator in the linked issue.

Reply:
The issue is about creating new analyzer for GetValueOrDefault and the analyzer could support the pattern too.

Reply:
The dotnet/runtime issue seems to be for the case `GetValueOrDefault()` not `GetValueOrDefault(defaultValue)`.

I think we should merge the changes as-is, and then consider a follow-up PR when new analyzer recommendations are available.

Reply:
I'd prefer to discuss this in .Net Runtime issue and postpone the PR. 
I really don't like the pattern for value types. :-)

Reply:
> I really don't like the pattern for value types. :-)

What is the problem?

[Documentation](https://docs.microsoft.com/dotnet/csharp/language-reference/operators/null-coalescing-operator) explicitly states the `??` operator can be used with nullable value types, and even uses `int` in an example.

Reply:
Personally I prefer the first form as well